jlm/rvsdg/node.cpp

```diff
@@ -252,8 +252,16 @@ node_output::node_output(Node * node, std::shared_ptr<const rvsdg::Type> type)
       node_(node)
 {}
 
+static Node::Id
+CreateNodeId() noexcept
+{
+  static Node::Id id = 0;
```

Comment:
Well this immediately causes thread-safety issues for the entire library (e.g. two threads using the rvsdg library concurrently, on _entirely_ distinct compilation units), so this is not advisable. I would suggest not making a mutable library-wide global. Per region should be fine.